qr-code/src/main/java/com/adyen/checkout/qrcode/QrCodeViewNew.kt

```diff
@@ -0,0 +1,147 @@
+
+    init {
+        orientation = VERTICAL
+        val padding = resources.getDimension(R.dimen.standard_double_margin).toInt()
+        setPadding(padding, padding, padding, padding)
+    }
+
+    override fun initView(delegate: ComponentDelegate, coroutineScope: CoroutineScope, localizedContext: Context) {
+        if (delegate !is QRCodeDelegate) throw IllegalStateException("Unsupported delegate type")
+
+        this.localizedContext = localizedContext
+        initLocalizedStrings(localizedContext)
+        imageLoader = ImageLoader.getInstance(context, delegate.configuration.environment)
+
+        observeDelegate(delegate, coroutineScope)
+
+        binding.copyButton.setOnClickListener { copyCode(delegate) }
+    }
+
+    private fun initLocalizedStrings(localizedContext: Context) {
+        binding.copyButton.setLocalizedTextFromStyle(R.style.AdyenCheckout_QrCode_CopyButton, localizedContext)
+    }
+
+    private fun observeDelegate(delegate: QRCodeDelegate, coroutineScope: CoroutineScope) {
+        delegate.outputDataFlow
+            .onEach { outputDataChanged(it) }
+            .launchIn(coroutineScope)
+
+        delegate.timerFlow
+            .onEach { onTimerTick(it) }
+            .launchIn(coroutineScope)
+    }
+
+    private fun outputDataChanged(outputData: QRCodeOutputData?) {
+        Logger.d(TAG, "onChanged")
+
+        if (outputData == null) return
+
+        updateMessageText(outputData.paymentMethodType)
+        updateLogo(outputData.paymentMethodType)
+    }
+
+    @StringRes
+    private fun getMessageTextResource(paymentMethodType: String?): Int? {
+        return when (paymentMethodType) {
+            PaymentMethodTypes.PIX -> R.string.checkout_qr_code_pix
+            else -> null
+        }
+    }
+
+    private fun updateLogo(paymentMethodType: String?) {
+        Logger.d(TAG, "updateLogo - $paymentMethodType")
+        if (!paymentMethodType.isNullOrEmpty()) {
+            imageLoader.load(paymentMethodType, binding.imageViewLogo)
+        }
+    }
+
+    private fun updateMessageText(paymentMethodType: String?) {
+        val resId = getMessageTextResource(paymentMethodType) ?: return
+        binding.textViewTopLabel.text = localizedContext.getString(resId)
+    }
+
+    private fun onTimerTick(timerData: TimerData) {
+        val minutes = TimeUnit.MILLISECONDS.toMinutes(timerData.millisUntilFinished)
+        val seconds = TimeUnit.MILLISECONDS.toSeconds(timerData.millisUntilFinished) % TimeUnit.MINUTES.toSeconds(1)
+
+        val minutesSecondsString = localizedContext.getString(
+            R.string.checkout_qr_code_time_left_format,
+            minutes,
+            seconds
+        )
+
+        binding.textViewTimer.text = localizedContext.getString(
+            R.string.checkout_qr_code_timer_text,
+            minutesSecondsString
+        )
+        binding.progressIndicator.progress = timerData.progress
+    }
+
+    private fun copyCode(delegate: QRCodeDelegate) {
```

Comment:
Makes more sense to pass the `qrCodeData` directly to this method, instead of the whole delegate

Reply:
Well, you wrote that method...

![image](https://user-images.githubusercontent.com/9079915/190596554-db39be40-cdbb-4f49-84d0-4f64c3686211.png)

Reply:
I know 😂 Just commented that as a reminder for myself